app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,21 @@
+class CommentsController < ApplicationController
+  def new
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new
+  end
+
+  def create
+    post = Post.find(params[:post_id])
+    Comment.create(post_params(post))
+    redirect_to user_post_path(current_user, post)
+  end
```

Comment:
- [ ] When creating your comment, you are not checking for potential errors using the `if else` statements as you did in your post_controller, i kindly suggest you do that so that incase there is an error while creating a comment, an error message can be displayed and handled properly, see the code snippet below as a guide
- code snippet
```ruby
@comment =  Comment.create(post_params(post))
 if @comment.save
      flash[:notice] = 'Comment created successfully'
      redirect_to user_post_path(current_user, post)
      else
      flash.now[:error] = "Error: Couldn't create comment"
      render :new, status: :unprocessable_entity
    end
```